lib/falcon/adapters/input.rb

```diff
@@ -81,6 +81,7 @@ def read(length = nil, buffer = nil)
 						@buffer = @buffer.byteslice(remaining_length..-1)
 					else
 						buffer << @buffer
+						@buffer.clear
```

Comment:
It surprises me this is necessary, but it makes sense when I think about it. Will this force Ruby to `free` the string?

Reply:
Yep, it didn't affect RSS, so I reverted it on `master` – https://github.com/socketry/falcon/commit/12b8818812b23c920e545e6b4c91e08e5348ee04